pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -142,3 +196,129 @@ func (a *GCPActuator) getImageID(cd *hivev1.ClusterDeployment, logger log.FieldL
 		return "", errors.New(msg)
 	}
 }
+
+// obtainLease uses the Hive MachinePoolNameLease resource to obtain a unique, single character
+// for use in the name of the machine pool. We are severely restricted on name lengths on GCP
+// and effectively have one character of flexibility with the naming convention originating in
+// the installer.
+func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, cd *hivev1.ClusterDeployment, logger log.FieldLogger) (leaseChar string, leaseCreated bool, leaseErr error) {
+	logger.Debugf("obtaining lease for pool: %s", pool.Name)
+
+	leases := &hivev1.MachinePoolNameLeaseList{}
+	err := a.client.List(context.TODO(), leases, client.InNamespace(pool.Namespace),
+		client.MatchingLabels(map[string]string{
+			constants.ClusterDeploymentNameLabel: cd.Name,
+			constants.MachinePoolNameLabel:       pool.Name,
```

Comment:
Just found a bug with this refactor here, in limiting only to this pool, we later use this list to calc all the leases in use and thus always find nothing, resulting in possibility of hitting a char already in use error.